pkg/cmd/pulumi/util.go

```diff
@@ -367,6 +369,10 @@ func chooseStack(ctx context.Context,
 	} else if len(options) == 0 {
 		// If no options are available, we can't offer a choice!
 		return nil, errors.New("this command requires a stack, but there are none")
+	} else {
+		if defaultOption == "" {
```

Comment:
```suggestion
	} else if defaultOption == "" && len(options) > 0 {
```
Nit: line 374 implicitly depends on the check on line 369. Adding the explicit check to this `else` case allows us to more closely couple our assumptions.

Reply:
Actually it can just be `else if defaultOption == ""` since we already checked that options is not empty